### PR TITLE
Rename Samples applications to TestApplications

### LINKED ISSUE #484: Change naming for Samples and Examples application
In this repo Sample application means application used in tests. Example application is app just to illustrate something, placed in different folder.

I hope you find a way to rephrase to differentiate the usage. Alternatively, we can try to change the code if you have better names :)

_Originally posted by @Kielek in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/483#discussion_r848395578_

Based on conversation with @theletterf,
We should change
Sample.* applications to TestApplication.* and all Example.* applications to Sample.*.

### PR DESCRIPTION
**Pull request by Kielek**
Partially fixes #484

Changes proposed in this pull request:
Rename Samples applications to TestApplications

Examples to Samples will be discussed later.

### CODE REVIEW

**Review by nrcventura — APPROVED** [review]
This makes the intention of these applications easier for me to understand.

File: build/nuke/Build.Steps.Windows.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ partial class Build
         .Unlisted()
         .After(CompileManagedTests)
         .OnlyWhenStatic(() => IsWin)
-        .Triggers(PublishIisSamples)
+        .Triggers(PublishIisTestApplications)
```
Comment (rajkumar-rangaraj) [inline build/nuke/Build.Steps.Windows.cs:103]: nit: Consider using PublishIISTestApplications.

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

**Review by pjanotti — APPROVED** [review]
LGTM

**Pull request merged**